pkg/async/schedule/aws/cloud_watch_scheduler.go

```diff
@@ -68,15 +70,15 @@ type cloudWatchScheduler struct {
 	metrics cloudWatchSchedulerMetrics
 }
 
-func getScheduleName(scheduleNamePrefix string, identifier admin.NamedEntityIdentifier) string {
-	hashedIdentifier := hashIdentifier(identifier)
+func getScheduleName(scheduleNamePrefix string, identifier core.Identifier) string {
+	hashedIdentifier := HashIdentifier(identifier)
```

Comment:
Inorder to avoid the complexity of backward compatibility , should we continue to support AWS schedule identifier using project,domain, name

And only for the new native scheduler user project,domain,name and version
That way even though the interface is passing core.Identifier 
only the native scheduler will use the version fields whereas aws will continue to use it without the version.
Let me know what you think

Reply:
i like this proposal!

Reply:
yes please, I think the unique id creation logic should reside with the actual scheduler